File: Assets/core/LuaScripts/luacode/systems/DelayCommandSystem.lua
local BaseSystem=require "system.BaseSystem"
local DelayCommandSystem=class("DelayCommandSystem",BaseSystem)

--延迟执行遍历时会因为删除添加对象导致的操作
--此系统要靠后顺序加入到entityManager中
function DelayCommandSystem:initialize(...)
	DelayCommandSystem.super.initialize(arg)
end

function DelayCommandSystem:Update(entityManager,deltaTime)
	DelayCommandSystem.super.Update(entityManager,deltaTime)

end

function DelayCommandSystem:Destroy()

end

return DelayCommandSystem

File: Assets/core/Game/ECS/Systems/FSM/States/DelayCommandSystem.cs

using Kernel.core;
using System.Collections.Generic;

namespace Kernel.game
{
	public class DelayCommandSystem : BaseSystem, ITick
	{
		private IList<BitSet> subscribedComponentMasks;
		private BitSet subscribedComponentsMask;

		public DelayCommandSystem(ECSWorld entityManager):base(entityManager)
		{
			InitMaskInfo();
		}

		protected void InitMaskInfo()
		{
			subscribedComponentMasks = new List<BitSet>
			{
				ECSFamily.GetKey<SinglationComponent>(),
				ECSFamily.GetKey<DataComponent>()
			};

			subscribedComponentsMask = subscribedComponentMasks[0] | subscribedComponentMasks[1];
		}

		public void Tick(float deltaTime)
		{
			//获取多个组件
			var result = new List<HashSet<IComponent>>();
			entityManager.FectchAll(subscribedComponentsMask, subscribedComponentMasks, result);

			//获取一个组件

			var result2= new List<IComponent>();
			entityManager.FectchAll(subscribedComponentsMask, result2);
		}
	}
}

